### PR TITLE
RELATED: RAIL-770 Unwrap executions responses

### PR DESCRIPTION
**Pull request by vojty**

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/315/ : SUCCESS in 11s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-docker-el7/289/ : SUCCESS in 1m 50s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/317/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-docker-el7/291/ : SUCCESS in 1m 57s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/318/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-docker-el7/292/ : SUCCESS in 46s

File: src/execution/execute-afm.ts [hunk 1]
```diff
@@ -114,17 +116,18 @@ export class ExecuteAfmModule {
...
-        });
+        return this.fetchExecutionResult(pollingUri, offset)
+            .then((executionResult: Execution.IExecutionResultWrapper) => {
+                if (executionResult === null) {
+                    return null;
+                }
+
+                const newResult = prevResult ? mergePageData(prevResult, executionResult) : executionResult;
+
+                const nextOffset = nextPageOffset(executionResult.executionResult.paging);
```
Comment (martinmosko) [inline src/execution/execute-afm.ts]: I would rather rename variable to executionResultWrapper and similar for other variables which contains IExecutionResultWrapper 

`executionResultWrapper.executionResult.paging`

File: src/execution/execute-afm.ts [hunk 2]
```diff
@@ -85,15 +85,17 @@ export class ExecuteAfmModule {
...
                 const offset = Array(dimensionality).fill(0); // offset holds information on dimensionality
                 const pollingUri = executionResponse.executionResponse.links.executionResult;
-                return this.getOnePage(pollingUri, offset).then((executionResult: Execution.IExecutionResult) => {
-                    return {
-                        executionResponse,
-                        executionResult
-                    };
-                });
+                return this.getOnePage(pollingUri, offset)
+                    .then((executionResult: Execution.IExecutionResultWrapper) => {
```
Comment (martinmosko) [inline src/execution/execute-afm.ts]: `executionResultWrapper` ? :)

File: src/execution/execute-afm.ts [hunk 3]
```diff
@@ -85,15 +85,17 @@ export class ExecuteAfmModule {
 
         return this.xhr.post(`/gdc/app/projects/${projectId}/executeAfm`, { body: JSON.stringify(execution) })
             .then((r => r.getData()))
-            .then((executionResponse) => {
+            .then((executionResponse: Execution.IExecutionResponseWrapper) => {
```
Comment (martinmosko) [inline src/execution/execute-afm.ts]: `executionResponseWrapper`

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/319/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-docker-el7/293/ : SUCCESS in 48s

File: src/execution/execute-afm.ts [hunk 4]
```diff
@@ -85,15 +88,17 @@ export class ExecuteAfmModule {
...
-                const pollingUri = executionResponse.executionResponse.links.executionResult;
-                return this.getOnePage(pollingUri, offset).then((executionResult: Execution.IExecutionResult) => {
-                    return {
-                        executionResponse,
-                        executionResult
-                    };
-                });
+                const pollingUri = executionResponseWrapper.executionResponse.links.executionResult;
+                return this.getOnePage(pollingUri, offset)
+                    .then((executionResult: Execution.IExecutionResultWrapper) => {
```
Comment (martinmosko) [inline src/execution/execute-afm.ts]: `executeResultWrapper`

File: src/execution/execute-afm.ts [hunk 5]
```diff
@@ -113,18 +118,23 @@ export class ExecuteAfmModule {
...
-
-            const newResult = prevResult ? mergePageData(prevResult, executionResult) : executionResult;
-
-            const nextOffset = nextPageOffset(executionResult.executionResult.paging);
-            return nextOffset
-                ? this.getOnePage(pollingUri, nextOffset, newResult)
-                : newResult;
-        });
+    private getOnePage(
+        pollingUri: string, offset: number[], prevResultWrapper?: Execution.IExecutionResultWrapper
```
Comment (martinmosko) [inline src/execution/execute-afm.ts]: there was for prevResultWrapper set a default value (null) its intentional change?

Comment (vojty) [reply src/execution/execute-afm.ts]: `Type 'null' is not assignable to type 'IExecutionResultWrapper'.` :)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/320/ : SUCCESS in 15s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-docker-el7/294/ : SUCCESS in 1m 28s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/322/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-docker-el7/296/ : SUCCESS in 1m 56s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/323/ : SUCCESS in 10s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-docker-el7/297/ : SUCCESS in 48s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/324/ : SUCCESS in 12s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-docker-el7/298/ : SUCCESS in 51s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://ci.intgdc.com/job/gooddata-js-merge-compliance-check/69/ : SUCCESS in 17s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-docker-el7/299/ : SUCCESS in 52s

**Pull request merged**